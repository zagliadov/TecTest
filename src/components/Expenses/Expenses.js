import React from "react";

export const Expenses = () => {
  return (
    <div className="flex flex-col">
      <h2 className="text-center py-4">Expenses:</h2>
    </div>
  );
};
