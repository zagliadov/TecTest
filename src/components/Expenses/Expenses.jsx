import React, { useState } from "react";
import { useEffect } from "react";
import { Search } from "./components/Search/Search";

const getUser = () => Promise.resolve({id: 1, name: "Daniil"})
export const Expenses = () => {
  const [search, setSearch] = useState("");
  const [user, setUser] = useState("");

  useEffect(() => {
    const loadUser = async () => {
      const user = await getUser();
      setUser(user);
    };
    loadUser();
  }, [])

  const handleSearch = ({ target }) => {
    setSearch(target.value);
  };

  return (
    <div className="flex flex-col">
      <h2 className="text-center py-4">Expenses:</h2>
      {user && <h2>Logged in as { user.name }</h2>}
      <Search value={search} onChange={handleSearch}>
        Search: 
      </Search>
      <p>Searches for {search ? search : "..."}</p>
    </div>
  );
};
