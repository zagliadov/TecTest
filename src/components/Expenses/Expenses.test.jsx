import { render, screen } from "@testing-library/react";
import { Expenses } from "./Expenses";

describe("Expenses component", () => {
  

  it("Expenses render", async () => {
    const { asFragment } = render(<Expenses />);
    expect(await screen.findByText(/Logged in as/i)).toBeInTheDocument();
    expect(asFragment(<Expenses />)).toMatchSnapshot();
  });
});
