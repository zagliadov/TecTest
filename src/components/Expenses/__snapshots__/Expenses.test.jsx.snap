// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses component Expenses render 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <h2
      class="text-center py-4"
    >
      Expenses:
    </h2>
    <h2>
      Logged in as Daniil
    </h2>
    <div>
      <img
        alt="search images"
        src=""
      />
      <label
        for="search"
      >
        Search:
      </label>
      <input
        id="search"
        placeholder="search text..."
        type="text"
        value=""
      />
    </div>
    <p>
      Searches for ...
    </p>
  </div>
</DocumentFragment>
`;
